src/Symfony/Component/VarDumper/Caster/ReflectionCaster.php

```diff
@@ -20,19 +20,221 @@
+
+        return $a;
+    }
+
+    public static function castClass(\ReflectionClass $c, array $a, Stub $stub, $isNested, $filter = 0)
+    {
+        $p = Caster::PREFIX_VIRTUAL;
+
+        if ($n = \Reflection::getModifierNames($c->getModifiers())) {
+            $a[$p.'modifiers'] = implode(' ', $n);
+        }
+
+        self::addMap($a, $c, array(
+            'extends' => 'getParentClass',
+            'implements' => 'getInterfaceNames',
+            'constants' => 'getConstants',
+        ));
+
+        foreach ($c->getProperties() as $n) {
+            $a[$p.'properties'][$n->name] = $n;
+        }
+
+        foreach ($c->getMethods() as $n) {
+            $a[$p.'methods'][$n->name] = $n;
+        }
+
+        if (!($filter & Caster::EXCLUDE_VERBOSE) && !$isNested) {
+            self::addExtra($a, $c);
+        }
 
         return $a;
     }
+
+    public static function castFunctionAbstract(\ReflectionFunctionAbstract $c, array $a, Stub $stub, $isNested, $filter = 0)
+    {
+        $p = Caster::PREFIX_VIRTUAL;
+
+        self::addMap($a, $c, array(
+            'returnsReference' => 'returnsReference',
+            'class' => 'getClosureScopeClass',
+            'this' => 'getClosureThis',
+        ));
+
+        if (isset($a[$p.'this'])) {
+            $a[$p.'this'] = new CutStub($a[$p.'this']);
+        }
+
+        foreach ($c->getParameters() as $v) {
+            $k = '$'.$v->name;
+            if ($v->isPassedByReference()) {
+                $k = '&'.$k;
+            }
+            if (method_exists($v, 'isVariadic') && $v->isVariadic()) {
+                $k = '...'.$k;
+            }
+            $a[$p.'parameters'][$k] = $v;
+        }
+
+        if ($v = $c->getStaticVariables()) {
+            foreach ($v as $k => &$v) {
+                $a[$p.'use']['$'.$k] =& $v;
+            }
+            unset($v);
+        }
+
+        if (!($filter & Caster::EXCLUDE_VERBOSE) && !$isNested) {
+            self::addExtra($a, $c);
+        }
+
+        return $a;
+    }
+
+    public static function castMethod(\ReflectionMethod $c, array $a, Stub $stub, $isNested)
+    {
+        $a[Caster::PREFIX_VIRTUAL.'modifiers'] = implode(' ', \Reflection::getModifierNames($c->getModifiers()));
+
+        return $a;
+    }
+
+    public static function castParameter(\ReflectionParameter $c, array $a, Stub $stub, $isNested)
+    {
+        $p = Caster::PREFIX_VIRTUAL;
+
+        self::addMap($a, $c, array(
+            'position' => 'getPosition',
+            'isVariadic' => 'isVariadic',
+            'byReference' => 'isPassedByReference',
+        ));
+
+        try {
+            if ($c->isArray()) {
+                $a[$p.'typeHint'] = 'array';
+            } elseif (method_exists($c, 'isCallable') && $c->isCallable()) {
+                $a[$p.'typeHint'] = 'callable';
+            } elseif ($v = $c->getClass()) {
+                $a[$p.'typeHint'] = $v->name;
+            }
```

Comment:
Looking at the PHP source, scalar typehints are not yet exposed through reflection because there is a pending discussion about the way to expose typehints in a unified way (same for return types). So it is not yet possible to handle this